src/arizona_template_renderer.erl

```diff
@@ -0,0 +1,320 @@
+-module(arizona_template_renderer).
+
+%% --------------------------------------------------------------------
+%% API function exports
+%% --------------------------------------------------------------------
+
+-export([client_render/2]).
+-export([server_render/2]).
```

Comment:
We can look at this later, but functions are usually verbs, unless e.g. there's an exception like setters and getters. In this case, for example, I'd interpret `client_render` as a getter for a "client" property called "render", but I'm pretty sure this wouldn't make sense.

**Note**: I'll let the API stabilise more before we dig into module/function names, as well as type names and their opacity.